src/Strategy/CommonClassesStrategy.php

```diff
@@ -72,7 +73,7 @@ final class CommonClassesStrategy implements DiscoveryStrategy
             ['class' => React::class, 'condition' => React::class],
         ],
         HttpClient::class => [
-            ['class' => SymfonyHttplug::class, 'condition' => SymfonyHttplug::class],
+            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, Psr17RequestFactory::class]],
```

Comment:
Won't this possibly break applications using HttpClient 4.3 where this was not necessary?

Reply:
The `HttplugClient` class that this is checking for doesn't exist in HttpClient 4.3 so it shouldn't be an issue.